### PR TITLE
fix: Include swiftshader directory when creating installer for Electron 10+

### PR DESCRIPTION
**Pull request by niik**
This is a continuation of https://github.com/electron/windows-installer/pull/367 with the intention of including the swiftshader directory (present in Electron 10+) as well as the `vk_swiftshader_icd.json`  file. These paths are required in order for electron to be able to run on systems where GPU acceleration is unavailable.

See https://github.com/electron/windows-installer/pull/367#issuecomment-762968779:

> 👋 We (the GitHub Desktop team) is looking into [upgrading to Electron 11](https://github.com/desktop/desktop/pull/11142) and came upon this issue which made us a tad bit nervous as it seems like it would affect Electon 11 as well (right?).
> 
> We'd be happy to help in any way we can to get this over the finish line but it's not immediately obvious how.
> 
> > I haven't looked in detail, but I suspect that this change will fail with older versions of Electron that do not contain the swiftshader files. This module doesn't currently specify the versions of Electron that it supports, so it effectively supports pretty much all versions. If we restrict that, this will require a major version bump. Otherwise the change will need to be fixed so that it works for versions without these files.
> 
> @malept Do you have a sense of what your preferred approach would be?
> 
> In case the approach would be to special case versions prior to Electron 10 do you envision having two different `nuspecttemplate` files and conditionally including one or the other?
> 
> https://github.com/electron/windows-installer/blob/7f6ea867cb54aaa2bb0a2973641198cc8e05a43a/src/index.ts#L107
> 
> Or would you rather attempt to add the entries to the existing template at runtime?
> 
> And last question (for now), what method would you prefer for detecting whether these files need to be included? Should we just check whether they exist in the `appDirectory` or would you like to specifically check the bundled electron version?

@malept Is the approach taken here in line with what you'd expect?

Closes #367.

### CODE REVIEW

**Comment by niik** [comment]
I believe https://github.com/electron/windows-installer/pull/376 will alllow the `test-mac` build to pass as well.

File: spec/installer-spec.ts [hunk 1]
```diff
@@ -38,4 +41,19 @@ test('creates a nuget package and installer', async (t): Promise<void> => {
...
+
+  if (process.platform !== 'win32') {
+    args.unshift(cmd);
+    const wineExe = process.arch === 'x64' ? 'wine64' : 'wine';
+    cmd = wineExe;
+  }
+
+  const packageContents = await spawn(cmd, args);
+  t.true(packageContents.includes('lib\\net45\\vk_swiftshader_icd.json'));
+  t.true(packageContents.includes('lib\\net45\\swiftshader\\libEGL.dll'));
```
Comment (malept) [inline spec/installer-spec.ts:58]: It would be nice if there was a test that didn't add the swiftshader files as well, to prevent regressions.

**Review by malept — CHANGES_REQUESTED** [review]
Thanks for working on this.

Can the fixtures be zero-sized like the others?

File: spec/installer-spec.ts (hunk 1 above)
Comment (niik) [reply spec/installer-spec.ts:58]: Agreed, I'll take a stab at it!

**Comment by niik** [comment]
> Can the fixtures be zero-sized like the others?

I just copied one of other binary fixtures here which I guess wasn't zero sized. Looking at the fixtures directory there seems to  be a mix of empty files and binary-like files. I'll empty them tho!

File: spec/installer-spec.ts (hunk 1 above)
Comment (niik) [reply spec/installer-spec.ts]: I encountered an issue where the two tests caused a race condition trying to delete the `Squirrel.exe` file in the `beforeEach` so I tweaked the approach to always copying the fixture app directory to a temporary location before packaging.

**Review by malept — COMMENTED** [review]
This looks fine now, I just have some suggestions about temporary directory naming.

File: spec/installer-spec.ts [hunk 2]
```diff
@@ -3,33 +3,38 @@ import path from 'path';
...
-const appDirectory = path.join(__dirname, 'fixtures/app');
+const fixtureAppDirectory = path.join(__dirname, 'fixtures/app');
 
-test.beforeEach(async (): Promise<void> => {
-  const updateExePath = path.join(appDirectory, 'Squirrel.exe');
+function spawn7z(args: string[]): Promise<string> {
+  const sevenZipPath = path.join(__dirname, '..', 'vendor', '7z.exe');
+  return process.platform !== 'win32'
+    ? spawn(process.arch === 'x64' ? 'wine64' : 'wine', [sevenZipPath, ...args])
+    : spawn(sevenZipPath, args);
```
[the remark this answers is not in the document]
Comment (niik) [reply spec/installer-spec.ts]: Sorry, I tried to stay true to the current coding style but old habits die hard, I'll extract the wine binary name into a variable.

**Comment by sergiou87** [comment]
Hi @malept!

Did you have a chance to take a look at @niik's changes? This PR is blocking desktop/desktop#11142, so please let me know if there is anything I can do to help getting this merged! 🙏 

Thank you! ❤️

**Review by malept — APPROVED** [review]
Thanks for your patience, this will go out in a new release shortly.

**Pull request merged**